src/main/ruby/core/truffle/regexp_operations.rb

```diff
@@ -19,5 +19,19 @@ def self.match(re, str, pos=0)
       pos = m.character_to_byte_index pos
       re.search_region(str, pos, str.bytesize, true)
     end
+
+    def self.last_match(a_binding)
+      Truffle::KernelOperations.frame_local_variable_get(:$~, a_binding)
+    end
+
+    def self.set_last_match(value, a_binding)
+      unless value.nil? || Truffle::Type.object_kind_of?(value, MatchData)
+        raise TypeError, "Wrong argument type #{value} (expected MatchData)"
+      end
+      Truffle::KernelOperations.frame_local_variable_set(:$~, a_binding, value)
+    end
```

Comment:
Is this code used somewhere else?
Otherwise I think it would make sense to have it in the lambdas above directly.

Reply:
It is/will be used elsewhere, because we can now get rid of the `regexp_set_block_last_match` and `regexp_set_last_match` primitives which are used in various string methods..

Reply:
OK.